x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_management/prebuilt_rules/rule_customization.cy.ts

```diff
@@ -0,0 +1,348 @@
+          typeIndexPatterns([getIndexPatterns()[0]]);
+          submitBulkEditForm();
+
+          waitForBulkEditActionToFinish({
+            updatedCount: 1,
+          });
+
+          expectToContainModifiedBadge('Non-customized prebuilt rule');
+        });
+
+        it('add tags', () => {
+          filterByElasticRules();
+          selectAllRules();
+
+          openBulkEditAddTagsForm();
+          typeTags(['custom-tag']);
+          submitBulkEditForm();
+
+          waitForBulkEditActionToFinish({
+            updatedCount: 1,
+          });
+
+          expectToContainModifiedBadge('Non-customized prebuilt rule');
+        });
+
+        it('delete tags', () => {
+          filterByElasticRules();
+          selectAllRules();
+
+          openBulkEditDeleteTagsForm();
+          typeTags([testTags[0]]);
+          submitBulkEditForm();
+
+          waitForBulkEditActionToFinish({
+            updatedCount: 1,
+          });
+
+          expectToContainModifiedBadge('Non-customized prebuilt rule');
+        });
+
+        it('add custom highlighted fields', () => {
+          filterByElasticRules();
+          selectAllRules();
+
+          openBulkEditAddInvestigationFieldsForm();
+          typeInvestigationFields(['host.name']);
+          submitBulkEditForm();
+
+          waitForBulkEditActionToFinish({
+            updatedCount: 1,
+          });
+
+          expectToContainModifiedBadge('Non-customized prebuilt rule');
+        });
+
+        it('delete custom highlighted fields', () => {
+          filterByElasticRules();
+          selectAllRules();
+
+          openBulkEditDeleteInvestigationFieldsForm();
+          typeInvestigationFields(['source.ip']);
+          submitBulkEditForm();
+
+          waitForBulkEditActionToFinish({
+            updatedCount: 1,
+          });
+
+          expectToContainModifiedBadge('Non-customized prebuilt rule');
+        });
+
+        it('modify rule schedules', () => {
+          filterByElasticRules();
+          selectAllRules();
+
+          clickUpdateScheduleMenuItem();
+
+          typeScheduleInterval('20');
+          setScheduleIntervalTimeUnit('Hours');
+
+          typeScheduleLookback('10');
+          setScheduleLookbackTimeUnit('Minutes');
+
+          submitBulkEditForm();
+
+          waitForBulkEditActionToFinish({
+            updatedCount: 1,
+          });
+
+          expectToContainModifiedBadge('Non-customized prebuilt rule');
+        });
+      });
+
+      describe('calculating the Modified badge', () => {
+        it('modified badge should appear on the rule details page when prebuilt rule is customized', function () {
+          patchRule('rule_1', { name: 'Customized prebuilt rule' }); // We want to make this a customized prebuilt rule
+          visitRulesManagementTable();
+
+          cy.get(RULE_NAME).contains('Customized prebuilt rule').click();
+          expectModifiedBadgeToBeDisplayed(); // Expect modified badge to be displayed
+        });
```

Comment:
I think for all these tests for the badge we should have a single, common setup that would create all the needed rules, and then in each test we'd be just opening the right pages and checking the badge. There's no need to re-setup each test from scratch.